eva-lib/src/test/java/uk/ac/ebi/eva/lib/repository/VariantEntityRepositoryTest.java

```diff
@@ -272,37 +287,81 @@ public void testFindByRegionsAndComplexFilters() {
 
         testFindByRegionsAndComplexFiltersHelper(regions, null, null, VariantEntityRepository.RelationalOperator.NONE,
                                                  null, VariantEntityRepository.RelationalOperator.NONE, null,
-                                                 VariantEntityRepository.RelationalOperator.NONE, null, 28);
+                                                 VariantEntityRepository.RelationalOperator.NONE, null, null, 28);
 
         regions = new ArrayList<>();
         regions.add(new Region("11", 180001, 180079)); //4
 
         testFindByRegionsAndComplexFiltersHelper(regions, null, null, VariantEntityRepository.RelationalOperator.NONE,
                                                  null, VariantEntityRepository.RelationalOperator.NONE, null,
-                                                 VariantEntityRepository.RelationalOperator.NONE, null, 4);
+                                                 VariantEntityRepository.RelationalOperator.NONE, null, null, 4);
 
         regions.add(new Region("11", 180150, 180180)); //5
 
         testFindByRegionsAndComplexFiltersHelper(regions, null, null, VariantEntityRepository.RelationalOperator.NONE,
                                                  null, VariantEntityRepository.RelationalOperator.NONE, null,
-                                                 VariantEntityRepository.RelationalOperator.NONE, null, 9);
+                                                 VariantEntityRepository.RelationalOperator.NONE, null, null, 9);
 
         regions.add(new Region("11", 180205, 180221)); //2
 
         testFindByRegionsAndComplexFiltersHelper(regions, null, null, VariantEntityRepository.RelationalOperator.NONE,
                                                  null, VariantEntityRepository.RelationalOperator.NONE, null,
-                                                 VariantEntityRepository.RelationalOperator.NONE, null, 11);
+                                                 VariantEntityRepository.RelationalOperator.NONE, null, null, 11);
+    }
+
+    @Test
+    public void testFindByRegionsAndComplexFiltersExclude() {
```

Comment:
Single and multiple field tests have been added, and run ok, but nested field exclusion aren't currently supported by the Fongo version that is used (https://github.com/fakemongo/fongo/commit/3f937e16dbc11604b3e9e97ef9e0a1ae9c598911- and the version that supports nested exclusion in Fongo does not support version 3 of the Mongo driver)